_docs/tasks/traffic-management-v1alpha3/fault-injection.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Fault Injection
+overview: This task shows how to inject delays and test the resiliency of your application.
+
+order: 20
+
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This task shows how to inject delays and test the resiliency of your application.
+
+## Before you begin
+
+* Setup Istio by following the instructions in the
+  [Installation guide]({{home}}/docs/setup/).
+
+* Deploy the [Bookinfo]({{home}}/docs/guides/bookinfo.html) sample application.
+
+* Initialize the application version routing by either first doing the
+  [request routing](./request-routing.html) task or by running following
+  commands:
+
+  ```bash
+  istioctl create -f samples/bookinfo/routing/route-rule-all-v1.yaml
+  istioctl replace -f samples/bookinfo/routing/route-rule-reviews-test-v2.yaml
+  ```
+
+# Fault injection
+
+## Fault injection using HTTP delay
+To test our Bookinfo application microservices for resiliency, we will _inject a 7s delay_
+between the reviews:v2 and ratings microservices, for user "jason". Since the _reviews:v2_ service has a
+10s timeout for its calls to the ratings service, we expect the end-to-end flow to
+continue without any errors.
+
+1. Create a fault injection rule to delay traffic coming from user "jason" (our test user)
+
+   ```bash
+   istioctl replace -f samples/bookinfo/routing/route-rule-ratings-test-delay.yaml
+   ```
+
+   Confirm the rule is created:
+
+   ```bash
+   istioctl get virtualservice ratings -o yaml
+   ```
+   ```yaml
+   apiVersion: networking.istio.io/v1alpha3
+   kind: VirtualService
+   metadata:
+     name: ratings
+     ...
+   spec:
+     hosts:
+     - ratings
+     http:
+     - fault:
+         delay:
+           fixedDelay: 7s
+           percent: 100
+       match:
+       - headers:
+           cookie:
+             regex: ^(.*?;)?(user=jason)(;.*)?$
+     - route:
```

Comment:
this is incorrect. route and fault are in the same object

Reply:
good catch. I didn't notice since it works anyway, because there's only one version of ratings running.

That said, I guess the route actually needs to be in both rules, in the first rule and also in the non-matched lower priority rule.